src/users/users.serializer.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from "@nestjs/common";
+
+import { AbstractBaseSerializer } from "@/common/serializers";
+import { TSerializationOptions } from "@/common/serializers/abstract-base-serializer.types";
+
+@Injectable()
+export class UsersSerializer extends AbstractBaseSerializer {
+  protected serializeOneOptions: TSerializationOptions = {
+    skipNull: true,
+    forceObject: true,
+    exclude: ["password"],
+    populate: ["teacher", "student"],
+  };
```

Comment:
[NIT] - Same comment. I think it is better to populate logically. You can use the serialize method at the controller level rather than this.